great_expectations/datasource/fluent/fluent_base_model.py

```diff
@@ -266,3 +276,20 @@ def _include_exclude_to_dict(
 
     def __str__(self):
         return self.yaml()
+
+
+def _recursively_set_config_value(
+    data: dict | list, config_provider: _ConfigurationProvider
+):
```

Comment:
Should we attach this to a class as a static method?

Reply:
Or make it a method on the ConfigurationProvider?

Reply:
Yes I struggled with were to put this.

Reply:
I don't want to `ConfigurationProvider` because this is only used for Fluent StyleDatasources.
I would make it a class method on the `ConfigStr` itself.